app/agreements/utils.ts

```diff
@@ -49,7 +49,8 @@ export function formatFraction(fraction: BigNumber) {
 }
 
 export function calculateAndFormatRatio(amount: string, neumarkBalance: string): string {
-  return new BigNumber(amount).div(new BigNumber(neumarkBalance)).toFixed(3);
+  const num = new BigNumber(amount).div(new BigNumber(neumarkBalance));
+  return num.isNaN() ? "0" : num.toFixed(3);
```

Comment:
I would prefer to trace it to the route of the problem. Is amount or neumarkBalance unparsable? and set it to "0" so it returns 0. If division by 0 is breaking it then just put if (neumarkBalance === "0") return "0" or something like that, please